### PR TITLE
Chaos testing

### PR DESCRIPTION
**Pull request by caesarxuchao**
Adding a chaos test where the apiserver is restarted many times in a 10-minute window.

/assign @yliaog

### CODE REVIEW

File: test/e2e/tests/chaos.go [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+func (t *StorageMigratorChaosTest) setupClients() {
+	cfg, err := clientcmd.BuildConfigFromFlags("", "/workspace/.kube/config")
+	if err != nil {
+		util.Failf("can't build client config: %v", err)
+	}
+	cfg = rest.AddUserAgent(cfg, "e2e test")
+	t.migrationClient, err = migrationclient.NewForConfig(cfg)
+	if err != nil {
+		util.Failf("can't build client: %v", err)
```
Comment (yliaog) [inline test/e2e/tests/chaos.go]: s/can't build client/can't build migration client/

File: test/e2e/tests/chaos.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+	}
+	defer ready()
+
+	t.Setup()
+	ready()
+	t.Test(sem.StopCh)
+}
+
+func apiserverRestartsFunc() {
+	args := []string{"compute", "--project", os.Getenv("PROJECT"), "ssh", "--zone", os.Getenv("KUBE_GCE_ZONE"), os.Getenv("CLUSTER_NAME") + "-master", "--command", "sudo pkill -9 kube-apiserver"}
```
Comment (yliaog) [inline test/e2e/tests/chaos.go]: kubelet will quickly restart the api server after it's pkill'ed, what about moving the apiserver manifest to somewhere, wait some period of time, then move it back? this way you can test your code works even after some period of apiserver downtime

File: test/e2e/tests/setup.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/kubernetes-sigs/kube-storage-version-migrator/test/e2e/util"
+)
+
+func setupMigrator() {
+	// setup the CRD to test.
+	// We install the CRD before the migrator so that the migrator sees the
+	// CRD in the first round of discovery.
+	testCRD := "../../test/e2e/crd.yaml"
+	// setup the migration system
+	crds := "../../manifests.local/crd.yaml"
```
[the remark this answers is not in the document]
Comment (caesarxuchao) [reply test/e2e/tests/setup.go]: It's generated after calling `make local-manifests`.

File: test/e2e/tests/chaos.go (hunk 2 above)
Comment (caesarxuchao) [reply test/e2e/tests/chaos.go]: Kubelet restarts the pod after an exponentially time off.

**Comment by caesarxuchao** [comment]
@yliaog PTAL. Thank you.

**Pull request merged**